app/models/member.rb

```diff
@@ -0,0 +1,78 @@
+class Member < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable, :omniauthable
+
+  has_many :blogs
+  has_many :blog_posts
+
+
+  # Convert username to friendly url format
+  def slug
+    if name.present?
+      name.downcase.gsub(" ", "-").parameterize
+    end
+  end
+
+  # Change default param for user from id to id-name for friendly urls.
+  # When finding in DB, Rails auto calls .to_i on param, which tosses
+  # name and doesn't cause any problems in locating user.
+  def to_param
+    if name.present?
+      "#{id}-#{name}"
+    end
+  end
+
+  ##### FOR OAUTH
+  TEMP_EMAIL_PREFIX = 'change@me'
+  TEMP_EMAIL_REGEX = /\Achange@me/
```

Comment:
We can change it if you want, its really just a placeholder while the omniauth person verifies their email addy, so won't be used for live accounts